src/functionsEmulator.js

```diff
@@ -64,25 +66,28 @@ FunctionsEmulator.prototype.stop = function() {
 };
 
 FunctionsEmulator.prototype._getPorts = function() {
-  var portsConfig = {
-    supervisorPort: this.options.port,
-    restPort: this.options.port + 1,
-  };
-  if (_.includes(this.options.targets, "hosting")) {
-    return _.mapValues(portsConfig, function(port) {
-      return port + 1; // bump up port numbers by 1 so hosting can be served on first port
-    });
-  }
-  return portsConfig;
+  return portfinder.getPortPromise().then((restPort) => {
+    var portsConfig = {
+      supervisorPort: this.options.port,
+      restPort,
+    };
+    if (_.includes(this.options.targets, "hosting")) {
+      return _.mapValues(portsConfig, function(port) {
+        return port + 1; // bump up port numbers by 1 so hosting can be served on first port
```

Comment:
This isn't guaranteed to be free as well, though, is it? Why not use a `portfinder.getPortPromise()` for this value as well?

Additionally, if a port is specified by the user, we should still use that port as the base and increment from there, IMO. Thoughts?